File: languages/qmljs/nodejsmodules/crypto.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-
# This file is part of qmljs, the QML/JS language support plugin for KDevelop
# Copyright (c) 2014 Denis Steckelmacher <steckdenis@yahoo.fr>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation; either version 2 of
# the License or (at your option) version 3 or any later version
# accepted by the membership of KDE e.V. (or its successor approved
# by the membership of KDE e.V.), which shall act as a proxy
# defined in Section 14 of version 3 of the license.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.




from jsgenerator import *
from common import *

# Print the license of the generated file (the same as the one of this file)
license()
basicTypes(globals())
require('buffer')

_object = 'new Object()'

Module().members(
    F(_array, 'getCiphers'),
    F(_array, 'getHashes'),
    F(_object, 'createCredentials', ('details', _object)),
    F('new Hash()', 'createHash', ('algorithm', _string)),
    Class('Hash').members(
        F(_void, 'update', ('data', _string), ('input_encoding', _string)),
        F(_string, 'digest', ('encoding', _string))
    ),
    F('new Hmac()', 'createHmac', ('algorithm', _string), ('key', _string)),
    Class('Hmac').members(
        F(_void, 'update', ('data', _string)),
        F(_string, 'digest', ('encoding', _string))
    ),
    F('new Cipher()', 'createCipher', ('algorithm', _string), ('password', _string)),
    F('new Cipher()', 'createCipheriv', ('algorithm', _string), ('password', _string), ('iv', _string)),
    Class('Cipher').members(
        F(_void, 'update', ('data', _string), ('input_encoding', _string), ('output_encoding', _string)),
        F(_string, 'final', ('output_encoding', _string)),
        F(_void, 'setAutoPadding', ('auto_padding', _bool))
    ),
    F('new Decipher()', 'createDecipher', ('algorithm', _string), ('password', _string)),
    F('new Decipher()', 'createDecipheriv', ('algorithm', _string), ('password', _string), ('iv', _string)),
    Class('Decipher').members(
        F(_void, 'update', ('data', _string), ('input_encoding', _string), ('output_encoding', _string)),
        F(_string, 'final', ('output_encoding', _string)),
        F(_void, 'setAutoPadding', ('auto_padding', _bool))
    ),
    F('new Sign()', 'createSign', ('algorithm', _string)),
    Class('Sign').members(
        F(_void, 'update', ('data', _string)),
        F(_string, 'sign', ('private_key', _string), ('output_format', _string))
    ),
    F('new Verify()', 'createVerify', ('algorithm', _string)),
    Class('Verify').members(
        F(_void, 'update', ('data', _string)),
        F(_bool, 'verify', ('object', _string), ('signature', _string), ('signature_format', _string))
    ),
    F('new DiffieHellman()', 'createDiffieHellman', ('prime_length', _int)),
    Class('DiffieHellman').members(
        F(_string, 'generateKeys', ('encoding', _string)),
        F(_string, 'computeSecret', ('other_public_key', _string), ('input_encoding', _string), ('output_encoding', _string)),
        F(_string, 'getPrime', ('encoding', _string)),
        F(_string, 'getGenerator', ('encoding', _string)),
        F(_string, 'getPublicKey', ('encoding', _string)),
        F(_string, 'getPrivateKey', ('encoding', _string)),
        F(_void, 'setPublicKey', ('public_key', _string), ('encoding', _string)),
        F(_void, 'setPrivateKey', ('private_key', _string), ('encoding', _string)),
    ),
    F('new DiffieHellman()', 'getDiffieHellman', ('group_name', _string)),
    F(_void, 'pbkdf2', ('password', _string), ('salt', _string), ('iterations', _int), ('keylen', _int), ('callback', 'function(){}')),
    F(_string, 'pbkdf2Sync', ('password', _string), ('salt', _string), ('iterations', _int), ('keylen', _int)),
    F('new buffer.Buffer()', 'randomBytes', ('size', _int), ('callback', 'function(){}')),
    F('new buffer.Buffer()', 'pseudoRandomBytes', ('size', _int), ('callback', 'function(){}')),
    Var(_string, 'DEFAULT_ENCODING')
).print()

File: plugins/qmljs/nodejsmodules/crypto.js
/*
    SPDX-FileCopyrightText: 2014 Denis Steckelmacher <steckdenis@yahoo.fr>

    SPDX-License-Identifier: GPL-2.0-only OR GPL-3.0-only OR LicenseRef-KDE-Accepted-GPL
*/

var buffer = require("buffer");

exports.getCiphers = function () { return []; };
exports.getCiphers();

exports.getHashes = function () { return []; };
exports.getHashes();

exports.createCredentials = function (details) { return new Object(); };
exports.createCredentials(new Object());

exports.createHash = function (algorithm) { return new Hash(); };
exports.createHash("");

/*
 * exports.Hash
 */
function Hash() { return ; }
exports.Hash = Hash;

exports.Hash.prototype.update = function (data, input_encoding) { return ; };
exports.Hash.prototype.update("", "");

exports.Hash.prototype.digest = function (encoding) { return ""; };
exports.Hash.prototype.digest("");


exports.createHmac = function (algorithm, key) { return new Hmac(); };
exports.createHmac("", "");

/*
 * exports.Hmac
 */
function Hmac() { return ; }
exports.Hmac = Hmac;

exports.Hmac.prototype.update = function (data) { return ; };
exports.Hmac.prototype.update("");

exports.Hmac.prototype.digest = function (encoding) { return ""; };
exports.Hmac.prototype.digest("");


exports.createCipher = function (algorithm, password) { return new Cipher(); };
exports.createCipher("", "");

exports.createCipheriv = function (algorithm, password, iv) { return new Cipher(); };
exports.createCipheriv("", "", "");

/*
 * exports.Cipher
 */
function Cipher() { return ; }
exports.Cipher = Cipher;

exports.Cipher.prototype.update = function (data, input_encoding, output_encoding) { return ; };
exports.Cipher.prototype.update("", "", "");

exports.Cipher.prototype.final = function (output_encoding) { return ""; };
exports.Cipher.prototype.final("");

exports.Cipher.prototype.setAutoPadding = function (auto_padding) { return ; };
exports.Cipher.prototype.setAutoPadding(true);


exports.createDecipher = function (algorithm, password) { return new Decipher(); };
exports.createDecipher("", "");

exports.createDecipheriv = function (algorithm, password, iv) { return new Decipher(); };
exports.createDecipheriv("", "", "");

/*
 * exports.Decipher
 */
function Decipher() { return ; }
exports.Decipher = Decipher;

exports.Decipher.prototype.update = function (data, input_encoding, output_encoding) { return ; };
exports.Decipher.prototype.update("", "", "");

exports.Decipher.prototype.final = function (output_encoding) { return ""; };
exports.Decipher.prototype.final("");

exports.Decipher.prototype.setAutoPadding = function (auto_padding) { return ; };
exports.Decipher.prototype.setAutoPadding(true);


exports.createSign = function (algorithm) { return new Sign(); };
exports.createSign("");

/*
 * exports.Sign
 */
function Sign() { return ; }
exports.Sign = Sign;

exports.Sign.prototype.update = function (data) { return ; };
exports.Sign.prototype.update("");

exports.Sign.prototype.sign = function (private_key, output_format) { return ""; };
exports.Sign.prototype.sign("", "");


exports.createVerify = function (algorithm) { return new Verify(); };
exports.createVerify("");

/*
 * exports.Verify
 */
function Verify() { return ; }
exports.Verify = Verify;

exports.Verify.prototype.update = function (data) { return ; };
exports.Verify.prototype.update("");

exports.Verify.prototype.verify = function (object, signature, signature_format) { return true; };
exports.Verify.prototype.verify("", "", "");


exports.createDiffieHellman = function (prime_length) { return new DiffieHellman(); };
exports.createDiffieHellman(1);

/*
 * exports.DiffieHellman
 */
function DiffieHellman() { return ; }
exports.DiffieHellman = DiffieHellman;

exports.DiffieHellman.prototype.generateKeys = function (encoding) { return ""; };
exports.DiffieHellman.prototype.generateKeys("");

exports.DiffieHellman.prototype.computeSecret = function (other_public_key, input_encoding, output_encoding) { return ""; };
exports.DiffieHellman.prototype.computeSecret("", "", "");

exports.DiffieHellman.prototype.getPrime = function (encoding) { return ""; };
exports.DiffieHellman.prototype.getPrime("");

exports.DiffieHellman.prototype.getGenerator = function (encoding) { return ""; };
exports.DiffieHellman.prototype.getGenerator("");

exports.DiffieHellman.prototype.getPublicKey = function (encoding) { return ""; };
exports.DiffieHellman.prototype.getPublicKey("");

exports.DiffieHellman.prototype.getPrivateKey = function (encoding) { return ""; };
exports.DiffieHellman.prototype.getPrivateKey("");

exports.DiffieHellman.prototype.setPublicKey = function (public_key, encoding) { return ; };
exports.DiffieHellman.prototype.setPublicKey("", "");

exports.DiffieHellman.prototype.setPrivateKey = function (private_key, encoding) { return ; };
exports.DiffieHellman.prototype.setPrivateKey("", "");


exports.getDiffieHellman = function (group_name) { return new DiffieHellman(); };
exports.getDiffieHellman("");

exports.pbkdf2 = function (password, salt, iterations, keylen, callback) { return ; };
exports.pbkdf2("", "", 1, 1, function(){});

exports.pbkdf2Sync = function (password, salt, iterations, keylen) { return ""; };
exports.pbkdf2Sync("", "", 1, 1);

exports.randomBytes = function (size, callback) { return new buffer.Buffer(); };
exports.randomBytes(1, function(){});

exports.pseudoRandomBytes = function (size, callback) { return new buffer.Buffer(); };
exports.pseudoRandomBytes(1, function(){});

exports.DEFAULT_ENCODING = "";

